src/__tests__/04b-lightning-payment.ts

```diff
@@ -303,6 +305,36 @@ functionToTests.forEach(({ fn, name, initialFee }) => {
     const { BTC: finalBalance } = await userWallet1.getBalances()
     expect(finalBalance).toBe(initBalance1)
   }, 60000)
+
+  it(`fails to pay above 2fa threshold without 2fa token`, async () => {
+    // this is needed because this test runs twice
+    if (!userWallet0.user.twoFactor.secret) {
```

Comment:
why do you check this as part of the test?

Reply:
Because this code block is only needed for the first run of this test, but not the second one (it will throw if we try to set 2fa secret if it's already present)

Reply:
maybe refactor the test to not run this here in that case?
it could be done in the test related to the user settings

Reply:
you could create an active virtual property from `userWallet0.user.twoFactor`. such that if `secret` is set, `active` return true

Reply:
Yep, I've created a `twoFactorEnabled` virtual on the schema, however that doesn't solve the issue here.
This test is primarily checks whether the payment is denied if a payment go over the 2FA limit without providing a valid 2fa token. For that we need the 2fa to be enabled on that particular user.

I could leave 2fa enabled from the user creation test, but that adds an ordering dependency, which is something we definitely do not want.

I can create a helper function that does the 2FA enabling part, but again, that won't allow us to remove this part from this test, it will just move the declaration to a different file and call that.